src/conduits/RwaOutputConduit3.t.sol

```diff
@@ -0,0 +1,533 @@
+        vm.startPrank(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-operator");
+        outputConduit.pick(address(0));
+    }
+
+    function testRevertOnPickAddressNotWhitelisted() public {
+        vm.expectRevert("RwaOutputConduit3/not-bud");
+        outputConduit.pick(vm.addr(1));
+    }
+
+    function testPush() public {
+        assertEq(outputConduit.to(), me);
+        assertEq(usdx.balanceOf(me), 0);
+        assertEq(usdx.balanceOf(address(outputConduit)), 0);
+        assertEq(dai.balanceOf(address(me)), 1000 ether);
+
+        dai.transfer(address(outputConduit), 500 ether);
+
+        assertEq(dai.balanceOf(me), 500 ether);
+        assertEq(dai.balanceOf(address(outputConduit)), 500 ether);
+
+        vm.expectEmit(true, true, false, false);
+        emit Push(address(me), 500 * USDX_BASE_UNIT);
+        outputConduit.push();
+
+        assertEq(usdx.balanceOf(address(me)), outputConduit.expectedGemAmt(500 ether));
+        assertEq(outputConduit.to(), address(0));
+    }
+
+    function testPushAmountWhenAlreadyHaveSomeGemBalanceGetExactAmount() public {
+        usdx.mint(100 * USDX_BASE_UNIT);
+        usdx.transfer(address(outputConduit), 100 * USDX_BASE_UNIT);
+
+        assertEq(outputConduit.to(), me);
+        assertEq(usdx.balanceOf(me), 0);
+        assertEq(usdx.balanceOf(address(outputConduit)), 100 * USDX_BASE_UNIT);
+        assertEq(dai.balanceOf(address(me)), 1000 ether);
+
+        dai.transfer(address(outputConduit), 500 ether);
+
+        assertEq(dai.balanceOf(me), 500 ether);
+        assertEq(dai.balanceOf(address(outputConduit)), 500 ether);
+
+        vm.expectEmit(true, true, false, false);
+        emit Push(address(me), 500 * USDX_BASE_UNIT);
+        outputConduit.push(500 ether);
+
+        assertEq(usdx.balanceOf(address(me)), outputConduit.expectedGemAmt(500 ether));
+        assertEq(usdx.balanceOf(address(outputConduit)), 100 * USDX_BASE_UNIT);
+        assertEq(outputConduit.to(), address(0));
+    }
+
+    function testPushAmountWhenAlreadyHaveSomeGemBalanceGetAll() public {
+        usdx.mint(100 * USDX_BASE_UNIT);
+        usdx.transfer(address(outputConduit), 100 * USDX_BASE_UNIT);
+
+        assertEq(outputConduit.to(), me);
+        assertEq(usdx.balanceOf(me), 0);
+        assertEq(usdx.balanceOf(address(outputConduit)), 100 * USDX_BASE_UNIT);
+        assertEq(dai.balanceOf(address(me)), 1000 ether);
+
+        dai.transfer(address(outputConduit), 500 ether);
+
+        assertEq(dai.balanceOf(me), 500 ether);
+        assertEq(dai.balanceOf(address(outputConduit)), 500 ether);
+
+        vm.expectEmit(true, true, false, false);
+        emit Push(address(me), 500 * USDX_BASE_UNIT);
+        outputConduit.push();
+
+        assertEq(usdx.balanceOf(address(me)), outputConduit.expectedGemAmt(500 ether));
+        assertEq(usdx.balanceOf(address(outputConduit)), 100 * USDX_BASE_UNIT);
+        assertEq(outputConduit.to(), address(0));
+    }
+
```

Comment:
Well I guess this is covered in the fuzz test, leaving the commit for you to consider if you want such test with hard coded values as well.
Feel free to resolve otherwise.

Reply:
Done